### ISSUE TITLE
Do not report use of import or export errors for ambient contexts

### ISSUE THREAD

**Issue by mhegazy** [Microsoft/TypeScript#9829]
```ts
// a.d.ts

export = a; //  error TS1148: Cannot use imports, exports, or module augmentations when '--module' is 'none'.
declare var a: number;
```

> tsc a.d.ts --m none

We should not report errors in this case.
